### PR TITLE
Toan nguyen26 js wip

### PR DESCRIPTION
**Pull request by Toan-Nguyen26**
Started to mov my js to Chatbase , still under progress

### CODE REVIEW

**Comment by Toan-Nguyen26** [comment]
More function was converted

**Comment by Toan-Nguyen26** [comment]
The promise was kinda done, not the best way to check for document ready but it's what I came up with

File: www/js/VisitorChat/5.0/ChatBase.js.php [hunk 1]
```diff
@@ -97,7 +97,8 @@
         this.serverURL = serverURL;
 
         if (typeof visitorchat_config == "object") {
-            $.extend(this.config, visitorchat_config);
+            extend(this.config, visitorchat_config);
```
Comment (ericras) [inline www/js/VisitorChat/5.0/ChatBase.js.php]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

File: www/js/VisitorChat/5.0/ChatBase.js.php [hunk 2]
```diff
@@ -574,7 +619,7 @@
     updateChatContainerWithHTML:function (selector, html, sendAlerts) {
         //$.parseHTML(html)[0].outerHTML is used to compare the rendered html (the browser can change quotes, etc)
         //It just makes the comparison more accurate (at the cost of a little speed)
-        if ($(selector).html() === $.parseHTML(html)[0].outerHTML) {
+        if (selector.innerHTML === $.parseHTML((html)[0].outerHTML)) {
```
Comment (ericras) [inline www/js/VisitorChat/5.0/ChatBase.js.php]: "selector" by itself probably doesn't work.  Do you mean "document.querySelector(selector)" ?

File: www/js/VisitorChat/5.0/ChatBase.js.php [hunk 3]
```diff
@@ -631,42 +686,51 @@
 
             if (e.which == 13 && !e.shiftKey) {
                 e.preventDefault();
+                // submit() and val('') is a jquery function imo
                 if (VisitorChat.chatStatus == 'LOGIN') {
-                  $('#visitorchat_clientLogin').submit();
+                   document.querySelector('visitorchat_clientLogin').submit();
```
Comment (ericras) [inline www/js/VisitorChat/5.0/ChatBase.js.php]: Missing the # on the id: #visitorchat_clientLogin

**Comment by Toan-Nguyen26** [comment]
Most of them have ajax func and some other that are still jquery

**Pull request closed without merging**